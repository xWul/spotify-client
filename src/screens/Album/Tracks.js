/* LIBS */
import React, { Component } from 'react'
import styled from 'styled-components'
/* API */
import SpotifyWrapper from '../../api/index'
import { TOKEN } from '../../api/config'

const Content = styled.div`
  background-color: rgba(0, 0, 0, 0.5 );
  display: flex;
  flex-wrap: wrap;
  align-items: flex-start;
  border: 2px solid #FFF;
  position: absolute;
  width: 100%;
  height: 100%;
  opacity: 1;
  transition: all 0.3s ease;
  z-index: 110;
`

const Close = styled.span`
  cursor: pointer;
  display: block;
  padding: 10px;
  float: right;
  color: #FFF;
  width: 100%;
  text-align: right;
  z-index: 120;
`
const List = styled.ul`
  width: 100%;
  height: 89%;
  position: absolute;
  overflow: auto;
  margin-top: 8%;
  background-color: #FFF;
`
const Card = styled.li`
  display: flex;
  align-items: center;
  padding: 5px;
  border-radius: 3px;
  img {
    width: 45px;
    height: 45px;
  }
  p {
    padding: 0px 5px;
  }
`
const Load = styled.p`
  color: #FFF;
  text-align: center;
  width: 100%;
`

const Track = ({ track }) => {
  const { name } = track
  return (
    <Card>
      <p>{name}</p>
    </Card>
  )
}

const TracksList = ({ tracks }) => {
  return (
    tracks.items
      ? <List>
        {
          tracks.items.map(track => {
            return <Track track={track} key={track.id} />
          })
        }
      </List>
      : <Load>No tracks found.</Load>
  )
}

const Loading = () => {
  return <Load>Loading...</Load>
}

class Tracks extends Component {
  state = { tracks: [], isLoading: false }

  componentDidMount() {
    const { id } = this.props
    this.setState({ isLoading: true })
    const spotify = new SpotifyWrapper({
      token: TOKEN
    })
    spotify.album.getAlbum(id)
      .then(response => {
        this.setState({ tracks: response.tracks, isLoading: false })
      })
  }

  render() {
    const { onClose } = this.props
    const { tracks, isLoading } = this.state
    return (
      <Content>
        <Close onClick={onClose}>Close</Close>
        {tracks && <TracksList
          tracks={tracks} />}
        {isLoading && <Loading />}
      </Content>
    )
  }
}

export default Tracks
