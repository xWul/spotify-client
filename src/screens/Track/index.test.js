import React from 'react'
import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import mockFetch from '../../api/mockFetch'
import Tracks from './index'

describe('Tracks', function () {
  test('Tracks renderer without crashing', () => {
    fetch = mockFetch({tracks: {items: []}})
    const component = renderer.create(
      <Router><Tracks /></Router>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
