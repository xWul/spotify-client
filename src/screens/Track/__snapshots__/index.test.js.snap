// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracks Tracks renderer without crashing 1`] = `
<main
  className="sc-kpOJdX jXGnPb"
>
  <div
    className="sc-iwsKbI lhnNTF"
  >
    <div
      className="sc-gZMcBi bLjqeM"
    >
      <div
        className="sc-gqjmRU jraKAo"
      />
    </div>
  </div>
  <div
    className="sc-dnqmqq haujyR"
  >
    <form
      className="sc-htoDjs eCfHdW"
      method="POST"
      onSubmit={[Function]}
    >
      <div
        className="sc-bxivhb eswVzf"
      >
        <div
          className="sc-bwzfXH cAkfzP"
        >
          <select
            className="sc-htpNat gHkFix"
            name="option"
            onChange={[Function]}
            selected="track"
            value="track"
          >
            <option
              value="artist"
            >
              Artists
            </option>
            <option
              value="album"
            >
              Albums
            </option>
            <option
              value="track"
            >
              Tracks
            </option>
          </select>
        </div>
      </div>
      <div
        className="sc-ifAKCX cZUZpM"
      >
        <input
          className="sc-EHOje ghCCNY"
          name="query"
          onChange={[Function]}
          placeholder="Search for Artists, Albums or Tracks..."
          type="text"
          value=""
        />
      </div>
      <div
        className="sc-bZQynM ehozbl"
      >
        <button
          className="sc-gzVnrw eanGCw"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</main>
`;
